tests/api/shelves/create.test.js

```diff
@@ -38,22 +27,55 @@ describe('shelves:create', () => {
     }
   })
 
-  it('should create shelf', async () => {
+  it('should create a shelf', async () => {
     const name = shelfName()
-    const listing = 'public'
     const color = '#123412'
-    const { shelf } = await authReq('post', endpoint, { name, listing, color })
+    const visibility = [ 'public' ]
+    const { shelf } = await authReq('post', endpoint, { name, color, visibility })
     shelf.name.should.equal(name)
-    shelf.listing.should.equal(listing)
+    shelf.visibility.should.deepEqual(visibility)
     shelf.color.should.equal(color)
   })
 
   it('should create shelf with items', async () => {
     const name = shelfName()
     const item = await createItem()
-    const { shelf } = await authReq('post', endpoint, { name, listing: 'public', items: [ item._id ] })
+    const { shelf } = await authReq('post', endpoint, { name, visibility: [], items: [ item._id ] })
     const res = await authReq('get', `/api/shelves?action=by-ids&ids=${shelf._id}&with-items=true`)
     const resShelf = res.shelves[shelf._id]
     resShelf.items.should.containEql(item._id)
   })
+
+  it('should create a shelf with friends-only visibility', async () => {
+    const visibility = [ 'friends' ]
+    const { shelf } = await authReq('post', endpoint, { name: shelfName(), visibility })
+    shelf.visibility.should.deepEqual(visibility)
+  })
+
+  it('should create a shelf with groups-only visibility', async () => {
+    const visibility = [ 'groups' ]
+    const { shelf } = await authReq('post', endpoint, { name: shelfName(), visibility })
+    shelf.visibility.should.deepEqual(visibility)
+  })
+
+  it('should create a shelf with group-specific visibility', async () => {
+    const { group, member } = await createGroupWithAMember()
+    const visibility = [ `group:${group._id}` ]
```

Comment:
when a non existing `group._id` is passed .ie:
```
          const visibility = [ 'group:00000000000000000000000000000000' ]
```

it returns an error `user is not in that group`
could be `group not found`

[non-blocking]

Reply:
If that's non-blocking, I would prefer to pass on it, as the current implementation of `validateVisibilityKeys` makes it impractical to make the difference between groups the user is not part of and groups that don't exist: it just checks if the requested id is in `userGroupsIds` list.